src/check_python_versions/sources/pyproject.py

```diff
@@ -0,0 +1,364 @@
+                _ret = True
+    return _ret
+
+
+def is_setuptools_toml(
+    table: TOMLDocument
+) -> bool:
+    """Utility method to know if pyproject.toml is for setuptool."""
+    _ret = False
+    if BUILD_SYSTEM in table:
+        if BUILD_BACKEND in table[BUILD_SYSTEM]:
+            if SETUPTOOLS in \
+                    table[BUILD_SYSTEM][BUILD_BACKEND]:
+                _ret = True
+        if REQUIRES in table[BUILD_SYSTEM]:
+            if list(filter(lambda x: SETUPTOOLS in x,
+                           table[BUILD_SYSTEM][REQUIRES])):
+                _ret = True
+
+    #  "[tool.setuptools] table is still in beta"
+    #  "These configurations are completely optional
+    #    and probably can be skipped when creating simple packages"
+    if TOOL in table:
+        if SETUPTOOLS in table[TOOL]:
+            _ret = True
+    return _ret
+
+
+def is_flit_toml(
+    table: TOMLDocument
+) -> bool:
+    """Utility method to know if pyproject.toml is for flit."""
+    _ret = False
+    if TOOL in table:
+        if FLIT in table[TOOL]:
+            _ret = True
+    if BUILD_SYSTEM in table:
+        if BUILD_BACKEND in \
+                table[BUILD_SYSTEM]:
+            if FLIT in \
+                    table[BUILD_SYSTEM][BUILD_BACKEND]:
+                _ret = True
+        if REQUIRES in \
+                table[BUILD_SYSTEM]:
+            if list(filter(lambda x: FLIT in x,
+                           table[BUILD_SYSTEM][REQUIRES])):
+                _ret = True
+    return _ret
+
+
+def _get_poetry_classifiers(
+    table: TOMLDocument
+) -> List[str]:
+    if TOOL not in table:
+        return []
+    if POETRY not in \
+            table[TOOL]:
+        return []
+    if CLASSIFIERS not in \
+            table[TOOL][POETRY]:
+        return []
+    return cast(List[str], table[TOOL][POETRY][CLASSIFIERS])
+
+
+def _get_setuptools_flit_classifiers(
+    table: TOMLDocument
+) -> List[str]:
+    if PROJECT not in table:
+        return []
+    if CLASSIFIERS not in \
+            table[PROJECT]:
+        return []
+    return cast(List[str], table[PROJECT][CLASSIFIERS])
+
+
+def _get_pyproject_toml_classifiers(
+    filename: FileOrFilename = PYPROJECT_TOML
+) -> List[str]:
+    _classifiers = []
+    table = load_toml(filename)
+    if is_poetry_toml(table):
+        _classifiers = _get_poetry_classifiers(table)
+    if is_setuptools_toml(table) or is_flit_toml(table):
+        _classifiers = _get_setuptools_flit_classifiers(table)
+
+    return _classifiers
+
+
+def _get_poetry_python_requires(
+    table: TOMLDocument
+) -> Optional[str]:
+    if TOOL not in table:
+        return None
+    if POETRY not in table[TOOL]:
+        return None
+    if DEPENDENCIES not in \
+            table[TOOL][POETRY]:
+        return None
+    if PYTHON not in \
+            table[TOOL][POETRY][DEPENDENCIES]:
+        return None
+    return cast(str, table[TOOL][POETRY][DEPENDENCIES][PYTHON])
```

Comment:
Um, I checked Poetry's documentation and `tools.poetry.dependencies.python` uses a different syntax (`^3.8` instead of `>= 3.8`), so this will not work right.